### PR TITLE
PromptComponent converted to functional

### LINKED ISSUE #8668: [UI] Convert all Class-based components to functional components
### Current Behavior
A number of components are class-based. 

### Desired Behavior
Convert all the Class-based components to new updated functional components and change the required functions accordingly.

### Implementation
Convert all the Class-based components to new updated functional components. It is important here to convert a class-based component to a functional component you may need to convert some functions as well. Check if the function is used in other components or not.

Also, make sure to use the `hooks` while converting them. For example, you need to covert `withNotify` with `useNotification` hook. Please check if a hook for functional components is available and use that.

### Acceptance Tests
Make sure the component you have converted should work the same as before and try to upload a video of the working component.

### List of components we need to convert 👇🏽 
- [ ] Header - `/ui/components/Header.js`
- [ ] LoadTestTimerDialog - `/ui/components/load-test-timer-dialog.js`
- [ ] MeshAdapterConfigComponent - `/ui/components/MeshAdapterConfigComponent.js`
- [ ] MesheryAdapterPlayComponent - `/ui/components/MesheryAdapterPlayComponent.js`
- [ ] MesheryChart - `/ui/components/MesheryChart.js`
- [ ] MesheryChartDialog - `/ui/components/MesheryChartDialog.js`
- [ ] MesheryNotification - `/ui/components/MesheryNotification.js`
- [ ] MesheryPlayComponent - `/ui/components/MesheryPlayComponent.js`
- [ ] MesheryResultDialog - `/ui/components/MesheryResultDialog.js`
- [ ] MesherySettings - `/ui/components/MesherySettings.js`
- [ ] MesherySettingsPerformanceComponent - `/ui/components/MesherySettingsPerformanceComponent.js`
- [ ] Navigator - `/ui/components/Navigator.js`
- [ ] PromptComponent - `/ui/components/PromptComponent.js`
- [ ] ReactSelectWrapper - `/ui/components/ReactSelectWrapper.js`
- [ ] User - `/ui/components/User.js`
- [ ] UserPreference - `/ui/components/UserPreferences.js`
- [ ] MesheryPerformanceComponent - `/ui/components/MesheryPerformance/index.js`
- [ ] Index - `/ui/pages/index.js`
- [ ] Settings - `/ui/pages/settings.js`
- [ ] SMIResults - `/ui/pages/smi_results.js`
- [ ] NewFilters - `/ui/pages/configuration/filters.js`
- [ ] RemoteExtension - `/ui/pages/extension/[...component].js`
- [ ] Connections - `/ui/pages/management/connections.js`
- [ ] Manage - `/ui/pages/management/index.js`
- [ ] UserPref - `/ui/pages/user/preferences.js`

---

### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)

- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🖥 [Contributing to Meshery UI](https://docs.meshery.io/project/contributing/contributing-ui)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by giteshsarvaiya**
**Notes for Reviewers**

This PR fixes #8668 
PromptComponent is converted from class based component to functional component
- [x] all the paths tested

 promptComponent testing is done for the paths: 
- [x] filters.js   (ui/components/filters.js)
- [x] meshconfigcomponent.js  (ui/components/MeshConfigComponents.js)
- [x] header.js  (ui/components/header.js)
- [x] mesherypatterns.js  (ui/components/mesheryPattern.js)
- [x] mesherysettings.js  (ref declared but not used) (ui/components/mesherySettings.js)
- [x] mesherysettingsenvbutton.js  (ui/components/mesherySettingsEnvButton.js)
- [x] navigatorextension.js  (doesnt use refs) (ui/components/navigatorExtensions.js)
- [x] index.js  (ui/components/connections/index.js)
- [x] index.js  (ui/components/lifecycle/environments/index.js)
- [x] index.js  (ui/components/lifecycle/workspaces/index.js)
- [x] performanceProfiles.js  (ui/components/mesheryPerformance/performanceprofiles.js)

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by giteshsarvaiya** [comment]
may I know in which path can I see the "header.js" component?

**Comment by Yashsharma1911** [comment]
@giteshsarvaiya lint failure, you can run `eslint --fix .` in ui directory to fix lint issues by eslint itself

**Comment by Yashsharma1911** [comment]
Nice descriptive PR thanks

**Comment by giteshsarvaiya** [comment]
Okay, Let me fix it and commit it.. Will be doing by tomorrow. Came to bed for sleep.

File: ui/components/PromptComponent.js [hunk 1]
```diff
@@ -87,146 +87,124 @@ const IconButtonWrapper = styled(IconButton)(({ theme }) => ({
...
+  const [show, setShow] = useState(false);
+  const [title, setTitle] = useState('');
+  const [subtitle, setSubtitle] = useState('');
+  const [options, setOptions] = useState([]);
+  const [isChecked, setIsChecked] = useState(false);
+  const [showCheckbox, setShowCheckbox] = useState(false);
+  const [showInfoIcon, setShowInfoIcon] = useState(null);
+  const [variant, setVariant] = useState(null);
+  // const modalRef = useRef();
+  console.log('1');
```
[the remark this answers is not in the document]
Comment (giteshsarvaiya) [reply ui/components/PromptComponent.js:100]: Ohh, my bad, yaa thanks

File: ui/components/connections/index.js [hunk 2]
```diff
@@ -1078,30 +1084,32 @@ function Connections(props) {
...
-          k8scontextID: connections[index].metadata?.id,
-        }).subscribe({
-          next: (res) => {
-            updateProgress({ showProgress: false });
-            if (res.resetStatus === 'PROCESSING') {
-              notify({ message: `Database reset successful.`, event_type: EVENT_TYPES.SUCCESS });
-            }
-          },
-          error: handleError('Database is not reachable, try restarting server.'),
+      if (meshSyncResetRef.current) {
```
[the remark this answers is not in the document]
Comment (giteshsarvaiya) [reply ui/components/connections/index.js:1087]: Makes sense of what you are saying, let me see re-check if it works fine without checking the condition, I will remove that.

File: ui/components/connections/index.js [hunk 3]
```diff
@@ -1003,64 +1003,70 @@ function Connections(props) {
...
+        if (response === 'Delete') {
+          // let bulkConnections = {}
+          // selected.data.map(({ index }) => {
+          //   bulkConnections = {
+          //     ...bulkConnections,
+          //     [connections[index].id]: CONNECTION_STATES.DELETED
+          //   };
+          // })
+          // const requestBody = JSON.stringify(bulkConnections);
+          // updateConnectionStatus(requestBody);
```
[the remark this answers is not in the document]
Comment (giteshsarvaiya) [reply ui/components/connections/index.js:1062]: That isnt commented by me though

**Comment by Yashsharma1911** [comment]
@giteshsarvaiya let's discuss this in todays meeting

**Comment by giteshsarvaiya** [comment]
> @giteshsarvaiya let's discuss this in todays meeting

Okayy!

**Comment by giteshsarvaiya** [comment]
> @giteshsarvaiya lint failure, you can run `eslint --fix .` in ui directory to fix lint issues by eslint itself

Hello Yash, Do you know to solve this error?

**Comment by giteshsarvaiya** [comment]
@Yashsharma1911 , pushed the proposed changes.

**Comment by leecalcote** [comment]
@giteshsarvaiya will you sign-off on your commits? https://github.com/meshery/meshery/pull/10344/checks?check_run_id=22080323812

**Comment by giteshsarvaiya** [comment]
yes sir, will do and update the PR.

**Pull request closed without merging**